pennylane/tape/tape.py

```diff
@@ -360,6 +375,7 @@ def __enter__(self):
     def __exit__(self, exception_type, exception_value, traceback):
         try:
             super().__exit__(exception_type, exception_value, traceback)
+            # if exception_type is None:
```

Comment:
I think this is a line we can keep in, though maybe leaving a comment as to its purpose.